.github/workflows/tag-check.yml

```
@@ -0,0 +1,53 @@
+name: Check tag is annotated & signed
+
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  verify-tag:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          fetch-tags: true
+      
+      - name: Force fetch tags
+        run: |
+          git fetch --tags --force --prune
+          git fetch origin "refs/tags/${GITHUB_REF_NAME}:refs/tags/${GITHUB_REF_NAME}"
+
+      - name: Check tag is annotated
+        shell: bash
+        run: |
+          set -euo pipefail
+          tag="${GITHUB_REF_NAME}"
+          ref="refs/tags/${tag}"
+          echo $ref
+
+          if ! git show-ref --verify --quiet "$ref"; then
+            echo "Tag ${tag} not found after checkout"
```

```suggestion
            echo "::error::Tag '${tag}' not found after checkout"
```
Similar to the error at the bottom.

---

https://github.com/Dasharo/coreboot/pull/767/commits/ddba2507ea558acf52ec1fb2f33bfcfd895558f4